src/main/java/co/cask/hydrator/python/transform/Py4jPythonExecutor.java

```diff
@@ -0,0 +1,230 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.python.transform;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.hydrator.common.script.ScriptContext;
+import org.apache.commons.io.FileUtils;
+import org.awaitility.Awaitility;
+import org.awaitility.core.ConditionTimeoutException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import py4j.GatewayServer;
+import py4j.Py4JException;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Executes python code using py4j python library.
+ * Which involves starting python process as a server and than connecting java client to it.
+ */
+public class Py4jPythonExecutor implements  PythonExecutor {
+  private static final int PY4J_INIT_POLL_INTERVAL = 1;
+  private static final int PY4J_INIT_TIMEOUT = 120;
+  private static final int PYTHON_PROCESS_PORT_FILE_CREATED_INTERVAL = 1;
+  private static final int PYTHON_PROCESS_PORT_FILE_CREATED_TIMEOUT = 120;
+  private static final int PYTHON_PROCESS_STOP_TIMEOUT = 30;
+  private static final int PYTHON_PROCESS_KILL_TIMEOUT = 10;
+  private static final String USER_CODE_PLACEHOLDER = "\\$\\{cdap\\.transform\\.function\\}";
+
+  private final PythonEvaluator.Config config;
+  private File pythonProcessOutputFile;
+  private Logger logger;
+  private Process process;
+  private GatewayServer gatewayServer;
+  private Py4jTransport py4jTransport;
+  private File transformTempDir;
+
+  public Py4jPythonExecutor(PythonEvaluator.Config config) {
+    this.config = config;
+  }
+
+  private File prepareTempFiles() throws IOException {
+    URL url = getClass().getResource("/pythonEvaluator.py");
+    String scriptText = new String(Files.readAllBytes(Paths.get(url.getPath())), StandardCharsets.UTF_8);
+    scriptText = scriptText.replaceAll(USER_CODE_PLACEHOLDER, config.getScript());
+
+    Path transformTempDirPath = Files.createTempDirectory("transform");
+
+    logger.debug("Tmp folder is {}", transformTempDirPath);
+
+    String transformTempDirString = transformTempDirPath.normalize().toString();
+    transformTempDir = new File(transformTempDirString);
+
+    pythonProcessOutputFile = new File(transformTempDirString, "output.txt");
+    pythonProcessOutputFile.createNewFile();
```

Comment:
You miss exception handling here. Please use an output from createNewFile method or use FileUtils to handle it.